src/core/snapshot.ts

```diff
@@ -26,7 +26,14 @@ export class Snapshot<E extends Element = Element> {
   }
 
   get firstAutofocusableElement() {
-    return this.element.querySelector("[autofocus]")
+    const inertDisabledOrHidden = "[inert], [disabled], [hidden], details:not([open]), dialog:not([open])"
```

Comment:
Might be worth adding `:disabled`?

```suggestion
    const inertDisabledOrHidden = "[inert], [disabled], [hidden], details:not([open]), dialog:not([open]), :disabled"
```

Reply:
Interesting catch!

What's the semantic difference between the attribute and the pseudo class?

Reply:
`[disabled]` would only select if the attribute is present. Inputs can be disabled if they're part of a disabled fieldset:

```html
<fieldset disabled>
  <input type="text" autofocus>
</fieldset>
```

```js
console.log(document.querySelectorAll('input[disabled]').length) // 0
console.log(document.querySelectorAll('input:disabled').length) // 1
```

Having said that you're using `.closest`, so maybe this isn't an issue... although this might need to be verified but I think `<form disabled>` _won't_ disable child inputs:

```html
<form disabled>
  <input type="text" autofocus>
</form>
```

^ I think this _might_ need to be autofocused.

Reply:
Thank you for clarifying! I've removed the check for `[disabled]`, and replaced it with `:disabled`.